x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -429,8 +429,8 @@ const RuleDetailsPageComponent = () => {
 
   return (
     <>
-      {hasIndexWrite != null && !hasIndexWrite && <NoWriteAlertsCallOut />}
-      {userHasNoPermissions(canUserCRUD) && <ReadOnlyCallOut />}
+      <NoWriteAlertsCallOut />
+      <ReadOnlyCallOut />
```

Comment:
Not sure about the naming.

`NoWriteAlertsCallOut` -> `AlertsReadOnlyCallOut`?
`NoWriteAlertsCallOut` -> `AlertsReadOnlyCallOutSwitcher`?

`ReadOnlyCallOut ` -> `RulesReadOnlyCallOut`?
`ReadOnlyCallOut ` -> `RulesReadOnlyCallOutSwitcher`?

Reply:
I'm inclined toward `ReadOnlyCallout`/`ReadOnlyAlertsCallout`; those seem the least ambiguous to me

Reply:
Renamed to `ReadOnlyAlertsCallout` and `ReadOnlyRulesCallout`. Thank you, looks clean to me!